test/kcapi-enc-test.sh

```diff
@@ -253,18 +253,20 @@ test_filein_fileout()
 
 	diff_file $ORIGPT $GENPT "FILEIN / FILEOUT enc test ($keysize bits)"
 
-	# FIXME: error in openssl?
 	local ptsize=$(stat -c %s $ORIGPT)
 	local fullblock=$((ptsize%16))
+	local extra_openssl_opts=""
 
 	if [ $fullblock -eq 0 ]
 	then
-		return
+		# OpenSSL uses PKCS#7 padding which adds an extra pad block in this case
+		# Disable PKCS#7 padding when input length is a multiple of block size
+		extra_openssl_opts="$extra_openssl_opts -nopad"
 	fi
 
 	eval opensslkey=\$OPENSSLKEY${keysize}
-	openssl enc -aes-$keysize-cbc -in $ORIGPT -out $GENCT.openssl -K $opensslkey -iv $IV
-	openssl enc -d -aes-$keysize-cbc -in $GENCT -out $GENPT.openssl -K $opensslkey -iv $IV
+	openssl enc -aes-$keysize-cbc -in $ORIGPT -out $GENCT.openssl -K $opensslkey -iv $IV $extra_openssl_opts
+	openssl enc -d -aes-$keysize-cbc -in $GENCT -out $GENPT.openssl -K $opensslkey -iv $IV $extra_openssl_opts
```

Comment:
Would it work also if we just passed `-nopad` unconditionally? Then the `$fullblock -eq 0` conditional could be just removed.

Reply:
The problem is that kcapi-enc always pads but only for input length that is not multiple of block length, furthermore, it doesn't use PKCS#7 and pads only to the next multiple of block size, meaning that there will be NO padding in case the input length is already multiple of block size.
In contrast, PKCS#7 pads always even if the input length is already multiple of block size, in that case a full block is added as padding. This can be disabled by -nopad but it also disable all paddings. At least i think this what happens, according to my investigation.

Reply:
I see. Then your solution is indeed optimal.

Reply:
Is there a reason for not using PKCS#7 padding scheme in here: https://github.com/smuellerDD/libkcapi/blob/ce5866e0e77aa0e54200ddcab6c3b76cb7093090/apps/kcapi-enc.c#L355

To be compatible with OpenSSL ?